src/wasm-linker.cpp

```diff
@@ -173,7 +180,7 @@ void Linker::emscriptenGlue(std::ostream& o) {
     void visitCallImport(CallImport* curr) {
```

Comment:
The `emscripten.h` header has an EM_ASM macro that allows inline JS to be written. It's turned into a call to the outside using `emscripten_asm_const()`, and the runtime then runs the JS that was written there.

In wasm, we can't overload on imports, each import has a specific function type. So we convert `emscripten_asm_const()` into `emscripten_asm_const_vii()` for example, where `vii` is the signature (void return, two int params).

The pass also saves the JS code itself, which is just a string in the data section. (In theory we could have the backend emit it as metadata so we didn't need to do that part.)